fecgov/openFEC
# Add cycles_has_financial /activity in committee history and form_category in filings.

## Summary (required)

- Resolves [#[4016]](https://github.com/fecgov/openFEC/issues/4016)
- Ref PR: [https://github.com/fecgov/openFEC/pull/4043](https://github.com/fecgov/openFEC/pull/4043)

_Include a summary of proposed changes._

1) add four new columns in committees.CommitteeHistoryView(ofec_committee_history_mv):
cycles_has_financial
last_cycle_has_financial
cycles_has_activity
last_cycle_has_activity
<img width="788" alt="cmte_history_new_cols" src="https://user-images.githubusercontent.com/24395751/67797231-d25e3980-fa57-11e9-9a0b-49e6bfcbe624.png">


2) add one new column (as filter): **form_category** in filings.FilingsView(ofec_filings_all_mv)
form_category description:
The forms filed are categorized based on the nature of the filing:
    - REPORT:     F3, F3X, F3P, F3L, F4, F5, F7, F13
    - NOTICE:      F5, F24, F6, F9, F10, F11
    - STATEMENT:  F1, F2
    - OTHER:    F1M, F8, F99, F12, FRQ
    
## How to test the changes locally
1) checkout branch
2) point dev database
3) pytest
4) ./manage.py runserver

Test local and prod to compare:
1) test **/committee/committee_id/history/cycle** endpoint. should return same result.
there are four new columns in local

https://api.open.fec.gov/v1/committee/C00409011/history/2006/?sort_null_only=false&election_full=true&sort_hide_null=false&sort_nulls_last=false&per_page=20&page=1&sort=-cycle&API_KEY=DEMO_KEY

http://127.0.0.1:5000/v1/committee/C00409011/history/2006/?sort_null_only=false&election_full=true&sort_hide_null=false&sort_nulls_last=false&per_page=20&page=1&sort=-cycle&API_KEY=DEMO_KEY

2) test **/filings/** endpoint on local for new filter: form_category=REPORT

"count": 1813695, (form_category=report)
http://127.0.0.1:5000/v1/filings/?sort_null_only=false&sort_hide_null=false&per_page=20&sort=-receipt_date&form_category=report&sort_nulls_last=false&page=1

"count": 1264024 (form_category=report, most_recent=true)
http://127.0.0.1:5000/v1/filings/?sort_null_only=false&sort_hide_null=false&per_page=20&most_recent=true&sort=-receipt_date&form_category=report&sort_nulls_last=false&page=1


"count": 242741, (form_category=statement)
http://127.0.0.1:5000/v1/filings/?sort_null_only=false&sort_hide_null=false&per_page=20&sort=-receipt_date&form_category=statement&sort_nulls_last=false&page=1

"count": 46083(form_category=statement, most_recent=true)
http://127.0.0.1:5000/v1/filings/?sort_null_only=false&sort_hide_null=false&per_page=20&most_recent=true&sort=-receipt_date&form_category=statement&sort_nulls_last=false&page=1


3) test **/committee/committee_id/filings/**   and **/filings?committee_id**
should return the same data, (we can think about to deprecated **/committee/committee_id/filings/** in the future)

"count": 10,
http://127.0.0.1:5000/v1/filings/?sort_null_only=false&sort_hide_null=false&per_page=20&most_recent=true&sort=-receipt_date&form_category=REPORT&committee_id=C00567057&sort_nulls_last=false&page=1


"count": 10,
http://127.0.0.1:5000/v1/committee/C00567057/filings/?sort_null_only=false&sort_hide_null=false&per_page=20&most_recent=true&sort=-receipt_date&form_category=REPORT&sort_nulls_last=false&page=1
